kerby-kerb/kerb-client/src/main/java/org/apache/kerby/kerberos/kerb/client/impl/AbstractInternalKrbClient.java

```diff
@@ -170,11 +171,19 @@ public SgtTicket requestSgt(KOptions requestOptions) throws KrbException {
                     tgsRequest = new TgsRequestWithTgt(context, sgtTicket);
                 }
             }
+        } else {
+            //This code is for the no-tgt case but should be double-checked
```

Comment:
Is this part required for your use-case? If not I think it could be left out.

Reply:
I think that trying to 'manage use-cases' is wrong (IMHO). I have a question, is there a way to request an sgt without tgt? (e.g., OTP, user/pass) and in that case, where is the principal name stored?  I have speculated CLIENT_PRINCIPAL. Or is the USE_TGT the only accepted case for requestSGT?

Reply:
At least, I'll leave an empty else block with a reminder .. for other possible use cases...

Reply:
I also improved in another comments the clientPrincipal "null" case.. please refer to commit comments for details